src/main/java/com/direwolf20/buildinggadgets/common/items/gadgets/GadgetDestruction.java

```diff
@@ -259,64 +250,50 @@ public static void anchorBlocks(EntityPlayer player, ItemStack stack) {
         }
     }
 
-    public static SortedSet<BlockPos> getArea(World world, BlockPos pos, EnumFacing incomingSide, EntityPlayer player, ItemStack stack) {
-        SortedSet<BlockPos> voidPositions = new TreeSet<>(Comparator.comparingInt(Vec3i::getX).thenComparingInt(Vec3i::getY).thenComparingInt(Vec3i::getZ));
-        int depth = getToolValue(stack, NBTKeys.GADGET_VALUE_DEPTH);
-        if (depth == 0)
-            return voidPositions;
-
-        BlockPos startPos = (getAnchor(stack) == null) ? pos : getAnchor(stack);
-        EnumFacing side = (getAnchorSide(stack) == null) ? incomingSide : getAnchorSide(stack);
-        List<EnumFacing> directions = assignDirections(side, player);
-        IBlockState stateTarget = !Config.GADGETS.GADGET_DESTRUCTION.nonFuzzyEnabled.get() || GadgetGeneric.getFuzzy(stack) ? null : world.getBlockState(pos);
+    public static IPositionPlacementSequence getClearingPositions(World world, BlockPos pos, EnumFacing incomingSide, EntityPlayer player, ItemStack stack) {
         if (GadgetGeneric.getConnectedArea(stack)) {
-            String[] directionNames = NBTKeys.GADGET_VALUES;
-            AxisAlignedBB area = new AxisAlignedBB(pos);
-            for (int i = 0; i < directionNames.length; i++)
-                area = area.union(new AxisAlignedBB(pos.offset(directions.get(i), getToolValue(stack, directionNames[i]) - (i == 4 ? 1 : 0))));
+            Region boundary = getClearingRegion(pos, incomingSide, player, stack);
+            HashSet<BlockPos> voidPositions = new HashSet<>();
+            int depth = getToolValue(stack, NBTKeys.GADGET_VALUE_DEPTH);
+            if (depth == 0)
+                return new SetBackedPlacementSequence(voidPositions, boundary);
+
+            BlockPos startPos = (getAnchor(stack) == null) ? pos : getAnchor(stack);
+            IBlockState stateTarget = !Config.GADGETS.GADGET_DESTRUCTION.nonFuzzyEnabled.get() || GadgetGeneric.getFuzzy(stack) ? null : world.getBlockState(pos);
+            addConnectedCoordinates(world, player, startPos, stateTarget, voidPositions,
```

Comment:
If BlockPos doesn't implement comparable, we might need to switch back to TreeSet's log(n) worst-case performance (which is why we changed that from *List* to TreeSet somewhere around 2.6.2)

Reply:
Why do we need the positions to be sorted in the first place?

Reply:
The algorithm perfoms a contains check in order not to visit positions multiple times or wind up in a cycle.

Reply:
An unsorted set can also achieve what you've said. I presume this is used for rendering so that there is no side-duplication?

Reply:
We used a SortedSet (TreeSet) as this guarantees O(log(n)) time for the contains check, other hash-based implementations may have an expected time of O(1+alpha) but may degrade to an O(n) worst-case for non-comparable objects. Therefore if we encounter this worst case to happen more often, we might want to fall back to TreeSet

Reply:
As I said we came from an unsorted list, which had an expected contains check in O(n)...